aiogram/contrib/fsm_storage/mongo.py

```diff
@@ -70,6 +75,11 @@ async def get_client(self) -> AsyncIOMotorClient:
                     logger = logging.getLogger("aiogram")
                     logger.warning("Run `pip install dnspython==1.16.0` in order to fix ConfigurationError. More information: https://github.com/mongodb/mongo-python-driver/pull/423#issuecomment-528998245")
                 raise e
+
+            if self._db_from_uri:
+                # Since the URI was provided directly, and this flag was specified,
+                # extract db_name from the given URI
+                self._db_name = self._mongo.get_default_database().name
```

Comment:
Maybe it's better to do this in `__init__`?

Reply:
Theoretically speaking - yes. But there is no `self._mongo` in `__init__`. Well, technically there is, but it is `None`.
There's a whole lot of things I would move around in this class, but I'm trying to be as unintrusive as possible here

Reply:
Why do we need any flags?
If db is provided directly - use is.
If not - use default db.

Reply:
@Olegt0rr because it *might* break someone's code, I guess :thinking:

Reply:
Exactly. Maybe someone out there is using it with a URL containing one db_name and explicitly providing a different one. Not very likely, but possible. But I do agree that it would be much cleaner without the flag

Reply:
Deprecations may help us not to collect bad practices for fear of breaking something :D

Reply:
So, what do you suggest? Should I break off the compatibility?

Reply:
Database name in the link may be used to make auth only and then user may use any database name available to authorized user. So I don't even understand motivation to make this PR at all. It brings something messing.